ReactWindows/ReactNative.Shared/UIManager/Events/KeyEvent.cs

```diff
@@ -0,0 +1,44 @@
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace ReactNative.UIManager.Events
+{
+    class KeyEvent : Event
+    {
+        public const string KeyDownEventString = "topKeyDown";
+        public const string KeyUpEventString = "topKeyUp";
+
+        private readonly int _key;
+
+        public KeyEvent(string eventName, int viewTag, int key)
+            : base(viewTag)
+        {
+            EventName = eventName;
+            _key = key;
+        }
+
+        public override string EventName
+        {
+            get;
+        }
+
+        public override bool CanCoalesce
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override void Dispatch(RCTEventEmitter eventEmitter)
+        {
+            var eventData = new JObject
+                {
+                    { "target", ViewTag },
+                    { "key", _key },
```

Comment:
>_key [](start = 29, length = 4)

How do these keys compare to the keys exposed by the web? We should probably standardize on a set of keys so we can have consistency between platforms (e.g. UWP, macOS, web)

Reply:
There are currently some inconsistencies between ReactNative and ReactJS already (think the arrow keys), and there are differences between UWP and WPF alone (the UWP key codes are closer to Web than WPF). Perhaps something could be done, but I'd be careful at this point.

Reply:
Arrow up/left/right/down key codes are different between ReactNative iOS/Android (19-22) and ReactJS (37-40). This peculiarity is not fixed at ReactXP level, hence consumers have to use "if reactnative" like constructs inside their code.

ReactNative UWP seems to align with the ReactJS/Web (though I haven't tested all keys). I had to map the arrows to the ReactNative iOS/Android values so we don't kill current functionality in an app I won't name (and maybe others), but I suspect a future ReactXP API change is needed to align everything to the Web codes, for ex.

ReactNative WPF (that we don't yet support over ReactXP) seems to be adding new codes (or maybe we're using the wrong codes), but I haven't planned on fixing that with this PR..

Reply:
Oh, and using the Keys constant gives a chance to RNW consumers to write cross UWP/WPF apps that can be oblivious of the platform, though:
- you can still hit Enter vs Return snafu
- may not be good enough in ReactXP context

Reply:
There are currently some inconsistencies between ReactNative and ReactJS already (think the arrow keys), and there are differences between UWP and WPF alone (the UWP key codes are closer to Web than WPF). Perhaps something could be done, but I'd be careful at this point.

---
In reply to: [156587889](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r156587889) [](ancestors = 156587889)